src/api/reference.md

```diff
@@ -1,83 +1,83 @@
+~ 如果 doctype 没有出现模板里（比如模板只渲染一个 HTML 片段），那么您可以在这里指定它。在一些需要控制自闭合标签和布尔值属性的代码样式的时候非常有用。您可以阅读 [doctype](../language/doctype.html#doctype-option) 的说明来了解更多细节。
 pretty
 ~ boolean | string
-~ Adds whitespace to the resulting HTML to make it easier for a human to read using `'  '` as indentation. If a string is specified, that will be used as indentation instead (e.g. `'\t'`). Defaults to `false`.
+~ 在输出的 HTML 里添加 `'  '` 这样的空格缩进来获得更好的代码可读性。如果这里指定的是一个字符串（比如 `'\t'`），那么将会使用它来作为控制缩进的字符。默认为 `false`。
 filters
 ~ object
-~ Hash table of [custom filters](../language/filters.html#custom-filters). Defaults to `undefined`.
+~ 存放 [自定义过滤器](../language/filters.html#custom-filters) 的哈希表。默认为 `undefined`。
 self
 ~ boolean
-~ Use a `self` namespace to hold the locals. It will speed up the compilation, but instead of writing `variable` you will have to write `self.variable` to access a property of the locals object. Defaults to `false`.
+~ 是否使用一个叫做 `self` 的命名空间来存放局部变量。这可以加速编译的过程，但是，相对于原来书写比如 `variable` 来访问局部变量，您将需要改为 `self.variable` 来访问它们。默认为 `false`。
 debug
 ~ boolean
-~ If set to `true`, the tokens and function body are logged to stdout.
+~ 当设置为 `true` 时，编译产生的函数代码会被记录到标准输出。
 compileDebug
 ~ boolean
-~ If set to `true`, the function source will be included in the compiled template for better error messages (sometimes useful in development). It is enabled by default unless used with [Express](https://expressjs.com/) in production mode.
+~ 当设置为 `true` 时，源代码会被包含在编译出来的模板函数中，用于提供更详实的错误信息（这在开发时会有用）。它默认是启用的，除非是在 [Express](https://expressjs.com/) 的生产环境模式中。
 globals
 ~ Array<string>
-~ Add a list of global names to make accessible in templates.
+~ 该数组用于向模板中添加全局对象的名字，编译器将保证它们不被局部作用域影响。
 cache
 ~ boolean
-~ If set to `true`, compiled functions are cached. `filename` must be set as the cache key. Only applies to `render` functions. Defaults to `false`.
+~ 当设置为 `true` 时，编译出来的函数会被缓存下来。此时必须填写 `filename` 选项作为缓存的索引字段。该选项仅用于 `render` 函数。默认为 `false`。
 inlineRuntimeFunctions
 ~ boolean
-~ Inline runtime functions instead of `require`-ing them from a shared version. For `compileClient` functions, the default is `true` so that one does not have to include the runtime. For all other compilation or rendering types, the default is `false`.
+~ 相对于使用 `require` 来获得公用的运行时函数，是否需要直接嵌入这些运行时函数。在 `compileClient` 函数里默认是 `true`，因此就不需要再手动包含那些函数（从而让其能在浏览器上运行）。在其他的 `compile` / `render` 系列函数中，默认是 `false`。
 name
 ~ string
-~ The name of the template function. Only applies to `compileClient` functions. Defaults to `'template'`.
+~ 模板函数的名称。仅用于 `compileClient` 函数。默认值是 `'template'`。
 ```
 
-## Methods
+## 方法 ~~ Methods
 
 ### pug.compile(source, ?options)
 
-Compile a Pug template to a function which can be rendered multiple times with different locals.
+把一个 Pug 模板编译成一个可多次使用、可传入不同局部变量渲染的函数。
 
 ```parameter-list
 source
 ~ string
-~ The source Pug template to compile
+~ 需要编译的 Pug 代码
 options
 ~ ?options
-~ An options object
+~ 存放选项的对象
 ```
 
-```parameter-list returns
+```parameter-list (returns)
```

Comment:
This change is updated from https://github.com/pugjs/pug-en/commit/001690996928b8546792679dcb0c285e43e92076